edb/edgeql/compiler/func.py

```diff
@@ -87,6 +87,13 @@ def compile_FunctionCall(
     else:
         matched_call = matched[0]
 
+    func = matched_call.func
+    func_name = func.get_shortname(env.schema)
+
+    if not ctx.session_mode and func.get_session_only(env.schema):
+        raise errors.QueryError(
+            f'{func_name}() requires a session', context=expr.context)
```

Comment:
Perhaps a clearer "the {func_name}() function cannot be called in a non-session context"?

Reply:
Updated wording